presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java

```diff
@@ -458,6 +462,24 @@ public static Slice varcharJsonExtract(@SqlType("varchar(x)") Slice json, @SqlTy
         return JsonExtract.extract(json, jsonPath.getObjectExtractor());
     }
 
+    @ScalarFunction(value="json_extract_jayway", hidden=true)
+    @LiteralParameters({"x", "y"})
+    @SqlNullable
+    @SqlType(StandardTypes.JSON)
+    public static Slice varcharJsonExtractJayway(@SqlType("varchar(x)") Slice json, @SqlType("varchar(y)") Slice jsonPath)
+            throws IOException
+    {
+        // handle null jsons the same way as current json_extract implementation
+        requireNonNull(json, "json is null");
+        try (DynamicSliceOutput dynamicSliceOutput = new DynamicSliceOutput(ESTIMATED_JSON_OUTPUT_SIZE)) {
```

Comment:
Why do we use `ESTIMATED_JSON_OUTPUT_SIZE`? @nezihyigitbasi seems like this is from your initial commit. Can you give more background? Thanks!

Reply:
`DynamicSliceOutput` constructor requires an initial estimated size (unless things have changed since the last time I looked at it), if we have a better estimate we should use that here. This is just an arbitrary initial estimate and `DynamicSliceOutput` will grow the internal slice as needed.

Reply:
Does it actually allocate that memory upfront? Is there a way to know the obj size after calling `com.jayway.jsonpath.JsonPath.read` and allocate the right amount then? What's the overhead of allocating a slice too small? Vs. What's the overhead of allocating a slide too large? If we can't get a better estimation at run time and have to choose a fixed value, 512 sounds too big as default.

Reply:
this size is also used in the original implementation to initialize the dynamic slice